### PR TITLE
Replace async awaitRedraw by sync window.paint(window.graphics), refactor Direct 3D and Metal redrawers

### PR DESCRIPTION
**Pull request by igordmn**

### CODE REVIEW

File: skiko/src/jvmMain/kotlin/org/jetbrains/skiko/context/Direct3DContextHandler.kt [hunk 1]
```diff
@@ -103,7 +96,9 @@ internal class Direct3DContextHandler(layer: SkiaLayer) : ContextHandler(layer)
 
     override fun rendererInfo(): String {
         return super.rendererInfo() +
-            "Video card: ${directXRedrawer.getAdapterName(device)}\n" +
-            "Total VRAM: ${directXRedrawer.getAdapterMemorySize(device) / 1024 / 1024} MB\n"
+            "Video card: ${directXRedrawer.getAdapterName()}\n" +
```
Comment (olonho) [inline skiko/src/jvmMain/kotlin/org/jetbrains/skiko/context/Direct3DContextHandler.kt:99]: `directXRedrawer.adapterName`?

**Comment by olonho** [comment]
Doesn't seem to work:
```
Changed renderer for org.jetbrains.skiko.SkiaLayer[,0,0,0x0,invalid,alignmentX=0.0,alignmentY=0.0,border=,flags=16777217,maximumSize=,minimumSize=,preferredSize=]: new value is METAL
Exception in thread "main" java.lang.IllegalStateException: Method should be called from AWT event dispatch thread
        at org.jetbrains.skiko.SkiaLayer.update$skiko(SkiaLayer.kt:283)
        at org.jetbrains.skiko.redrawer.MetalRedrawer.update(MetalRedrawer.kt:59)
        at org.jetbrains.skiko.redrawer.MetalRedrawer.redrawImmediately(MetalRedrawer.kt:53)
        at org.jetbrains.skiko.SkiaLayer.paint(SkiaLayer.kt:193)
        at SkijaInjectSample.AppKt.createWindow(App.kt:116)
        at SkijaInjectSample.AppKt.main(App.kt:24)
        at SkijaInjectSample.AppKt.main(App.kt)
<==========---> 80% EXECUTING [1m 21s]
> :run
```

File: skiko/src/jvmMain/kotlin/org/jetbrains/skiko/context/Direct3DContextHandler.kt (hunk 1 above)
Comment (igordmn) [reply skiko/src/jvmMain/kotlin/org/jetbrains/skiko/context/Direct3DContextHandler.kt]: Fixed

**Comment by igordmn** [comment]
> Doesn't seem to work:

Fixed

**Review by olonho — APPROVED** [review]

**Pull request merged**